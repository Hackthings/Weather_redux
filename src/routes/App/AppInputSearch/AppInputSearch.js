import React, { Component } from 'react';
import Loader from '../../../components/Loader'

class AppInputSearch extends Component {

    constructor(props) {
        super(props)
    }

    parseCitiesToComponents(searchedCities) {
        if (!searchedCities) {
            return
        }
        const listItems = searchedCities.map( city =>
            <li key={city.name + city.stateAbbreviation}>
                {city.name} {city.stateAbbreviation}
            </li>
        );
        return (
            <ul>{listItems}</ul>
        );
    }

    render() {
        const { value, handleInputChange, searchedCities, isRequesting } = this.props;
        const parsedCities = this.parseCitiesToComponents(searchedCities);


        console.log(searchedCities);
        return (
            <div>
                <input type="text" value={value} onChange={handleInputChange} />
                {isRequesting && (<Loader />)}
                {!isRequesting && parsedCities}

            </div>
        )
    }
}


export default AppInputSearch