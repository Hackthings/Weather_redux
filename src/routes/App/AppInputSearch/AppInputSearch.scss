.AppInputSearch {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    max-width: 800px;
    margin: 0 auto;
    &__input {
        width: 100%;
        height: 45px;
        padding: 10px 15px;
        font-family: 'Lato', sans-serif;
        color: #00394c;
        font-size: 22px;
        border: none;
        border-radius: 0;
        outline-color: #0c4152;
        margin-bottom: 30px;
        &::placeholder {
            font-size: 18px;
            color: #cccccc;
            font-weight: 300;
        }
    }
}