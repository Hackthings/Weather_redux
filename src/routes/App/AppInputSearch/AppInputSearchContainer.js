import React, { Component } from 'react';
import AppInputSearch from './AppInputSearch';
import _ from 'lodash';

const url = 'http://autocomplete.wunderground.com/aq?query=';

class AppInputSearchContainer extends Component {

    constructor() {
        super();

        this.state = {
            searchedCities: null,
            isRequesting: true,
            value: ''
        };

        this.requestSuggestions = _.debounce(this.requestSuggestions, 1000);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        this.setState({
            value: event.target.value
        });

        const cityName = this.state.value;
        this.requestSuggestions(cityName);
    }

    async requestSuggestions(city) {
        const requestUrl = url + city;

        this.setState({
            isRequesting: true
        });

        try {
            const response = await fetch(requestUrl);
            const parseResponse = await response.json();
            const parsed = this.parseResponse(parseResponse.RESULTS);

            this.setState({
                searchedCities: parsed,
                isRequesting: false
            })
        }
        catch (error) {
            console.log('there was an error', error)
        }
    }

    parseResponse(response) {
        const searchedCities = [];

        response.map(element => {
            searchedCities.push({
                name: element.name,
                stateAbbreviation: element.c
            })
        });

        return searchedCities;
    }

    render() {
        const {
            searchedCities,
            value,
            isRequesting


        } = this.state;

        return (
            <AppInputSearch
                searchedCities={searchedCities}
                value={value}
                handleInputChange={this.handleChange}
                isRequesting={isRequesting} />

        )
    }
}

export default AppInputSearchContainer