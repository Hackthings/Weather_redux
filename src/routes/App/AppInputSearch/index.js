import './AppInputSearch.scss';

export {default} from './AppInputSearchContainer'