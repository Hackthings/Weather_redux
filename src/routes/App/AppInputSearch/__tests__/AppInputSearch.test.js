import React from 'react';
import AppInputSearch from '../AppInputSearch';
import { shallow, mount } from 'enzyme';


describe('<AppInputSearch/>', () => {
    it('should render Loader', () => {

        const props = {
            value: 'test1',
            handleInputChange: 'test2',
            searchedCities: [],
            isRequesting: true
        }
        const wrapper = shallow(<AppInputSearch {...props} />);

        expect(wrapper).toMatchSnapshot('isRequesting=true');
        wrapper.setProps({ isRequesting: false });
        expect(wrapper).toMatchSnapshot('isRequesting=false');

    });

})

describe('createPath() return path', () => {

    it('should createPath()', () => {

        const wrapper = shallow(<AppInputSearch />);
        const path = wrapper.instance().createPath('Name City, trash', 'AB');

        expect(path).toMatch(/\/city\/[A-Z]{2}\/[A-Za-z_]{2,}/);
        //console.log('path', path)
    });

})

describe('parseCitiesToComponents() return ul', () => {

    it('return ul', () => {

        const wrapper = shallow(<AppInputSearch />);
        const array = [
            {
                name: 'test1',
                stateAbbreviation: 'TJ'
            },
            {
                name: 'test2',
                stateAbbreviation: 'TD'
            },
            {
                name: 'test3',
                stateAbbreviation: 'TT'
            }
        ]

        const list = wrapper.instance().parseCitiesToComponents(array);

        expect(list).toMatchSnapshot();
       
    });

})