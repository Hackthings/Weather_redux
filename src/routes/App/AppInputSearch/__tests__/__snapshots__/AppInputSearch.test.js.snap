// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppInputSearch/> should render Loader: isRequesting=false 1`] = `
<div
  className="AppInputSearch"
>
  <input
    className="AppInputSearch__input"
    onChange="test2"
    type="text"
    value="test1"
  />
  <ul />
</div>
`;

exports[`<AppInputSearch/> should render Loader: isRequesting=true 1`] = `
<div
  className="AppInputSearch"
>
  <input
    className="AppInputSearch__input"
    onChange="test2"
    type="text"
    value="test1"
  />
  <Loader />
</div>
`;

exports[`parseCitiesToComponents() return ul return ul 1`] = `
<ul>
  <Link
    replace={false}
    to="/city/TJ/test1"
  >
    <li>
      <Button>
        test1
         
        TJ
      </Button>
    </li>
  </Link>
  <Link
    replace={false}
    to="/city/TD/test2"
  >
    <li>
      <Button>
        test2
         
        TD
      </Button>
    </li>
  </Link>
  <Link
    replace={false}
    to="/city/TT/test3"
  >
    <li>
      <Button>
        test3
         
        TT
      </Button>
    </li>
  </Link>
</ul>
`;
